qutebrowser/browser/mhtml.py

```diff
@@ -0,0 +1,383 @@
+# vim: ft=python fileencoding=utf-8 sts=4 sw=4 et:
+
+# Copyright 2015 Daniel Schadt
+#
+# This file is part of qutebrowser.
+#
+# qutebrowser is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# qutebrowser is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with qutebrowser.  If not, see <http://www.gnu.org/licenses/>.
+
+"""Utils for writing a MHTML file."""
+
+import functools
+import io
+import os
+import re
+
+import collections
+import uuid
+from email import policy, generator, encoders
+from email.mime import multipart
```

Comment:
There are still some `from`-imports here - I'd prefer `import email` and `import email.mime`, and then using `email.policy`, `email.mime.multipart`, etc.

Sorry for being so pedantic about this, but when I see `foo()` I always assume it's a function defined in this module, so this trips me up.

(also, no blank line needed before those imports)